resources/seccomp/aarch64-unknown-linux-musl.json

```diff
@@ -206,16 +206,7 @@
             },
             {
                 "syscall": "madvise",
-                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads",
-                "args": [
-                    {
-                        "index": 2,
-                        "type": "dword",
-                        "op": "eq",
-                        "val": 4,
-                        "comment": "libc::MADV_DONTNEED"
-                    }
-                ]
+                "comment": "Used by the VirtIO balloon device and by musl for some customer workloads. It is also used by aws-lc during random number generation. They setup a memory page that mark with MADV_WIPEONFORK to be able to detect forks. They also call it with -1 to see if madvise is supported in certain platforms." 
```

Comment:
I'm not 100% sure on how the seccompiler/bpf thingy works, but I'm assuming it would just be tedious to specify the exact parameters with which madvise would be called, and the syscall isn't spicy enough for us to care about that? :o

Reply:
Problem is they are trying madvise with `-1` which as when unsigned is `0xFFFFFFFF`, so unfortunately we can't distinguish.

Reply:
Would it be possible to codify this use-case in a test of some sort, just so that the restriction isn't re-added to the seccomp rules accidentally?

Reply:
I think that if the seccomp filter is changed to be stricter in the future the tests for `virtio-rng` will just break, because they will eventually exercise this particular `madvise` path. Do you have any suggestions for a more specific test?

Reply:
Would the aws-lc related tests fail now if the `libc::MADV_DONTNEED` rule was present? If not, is there anything we can do in with aws-lc to force this madvise path?

Reply:
they would! They always call `madvise` with -1: https://github.com/aws/aws-lc/blob/987bd7c672e2c3dd1cc76be3ce9eece93ba9bd88/crypto/fipsmodule/rand/fork_detect.c#L53, so the test would fail.